library/src/main/java/bio/terra/stairway/Control.java

```diff
@@ -0,0 +1,583 @@
+import java.sql.Connection;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Timestamp;
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
+import java.util.stream.Collectors;
+import javax.sql.DataSource;
+
+@SuppressFBWarnings(
+    value = "RCN_REDUNDANT_NULLCHECK_WOULD_HAVE_BEEN_A_NPE",
+    justification = "Spotbugs doesn't understand resource try construct")
+public class Control {
+  private final Stairway stairway;
+  private final DataSource dataSource;
+  private final FlightDao flightDao;
+  private final StairwayInstanceDao stairwayInstanceDao;
+
+  Control(
+      Stairway stairway,
+      DataSource dataSource,
+      FlightDao flightDao,
+      StairwayInstanceDao stairwayInstanceDao) {
+    this.stairway = stairway;
+    this.dataSource = dataSource;
+    this.flightDao = flightDao;
+    this.stairwayInstanceDao = stairwayInstanceDao;
+  }
+
+  // -- Flight methods --
+
+  private static final String flightSelectFrom =
+      " flightid, submit_time, class_name, completed_time, status,"
+          + "serialized_exception, stairway_id FROM flight ";
+
+  private static final String flightOrderPage =
+      " ORDER BY submit_time DESC OFFSET :offset LIMIT :limit";
+
+  private int countQuery(NamedParameterPreparedStatement statement) throws SQLException {
+    int flightCount = 0;
+    try (ResultSet rs = statement.getPreparedStatement().executeQuery()) {
+      while (rs.next()) {
+        flightCount = rs.getInt("total");
+      }
+    }
+    return flightCount;
+  }
+
+  public int countFlights(String status) throws SQLException {
+    StringBuilder sb = new StringBuilder();
+    sb.append("SELECT COUNT(*) as total FROM flight");
+    if (status != null) {
+      try {
+        FlightStatus.valueOf(status);
+      } catch (IllegalArgumentException ex) {
+        throw new IllegalArgumentException(
+            "Invalid status value. Values are "
+                + Arrays.stream(FlightStatus.values())
+                    .map(FlightStatus::toString)
+                    .collect(Collectors.joining(", ")));
+      }
+      sb.append(" WHERE status = :status");
+    }
+
+    try (Connection connection = dataSource.getConnection();
+        NamedParameterPreparedStatement statement =
+            new NamedParameterPreparedStatement(connection, sb.toString())) {
+      DbUtils.startReadOnlyTransaction(connection);
+      if (status != null) {
+        statement.setString("status", status);
+      }
+
+      int flightCount = countQuery(statement);
+      DbUtils.commitTransaction(connection);
+      return flightCount;
+    }
+  }
+
+  public int countOwned() throws SQLException {
+    final String sql = "SELECT COUNT(*) AS total FROM flight WHERE stairway_id IS NOT NULL";
+    try (Connection connection = dataSource.getConnection();
+        NamedParameterPreparedStatement statement =
```

Comment:
This is a case where I think `var` can be used to make the code shorter (and therefore easier to read) without sacrificing clarity. 
```
var statement =
```